openssl-dynamic/src/main/c/ssl.c

```diff
@@ -664,7 +664,7 @@ TCN_IMPLEMENT_CALL(jint, SSL, initialize)(TCN_STDARGS, jstring engine)
     }
 
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
-    if (SSLeay() < 0x0090700L) {
+    if (OpenSSL_version_num() < 0x0090700L) {
```

Comment:
FYI: OpenSSL_version_num is the same as `SSLeay` but the "new API". If an old version of OpenSSL is used we delegate internally.